.drone.yml

```diff
@@ -413,19 +475,35 @@ platform:
   arch: amd64
 
 steps:
-  - name: validate_go_mods
-    image: rancher/dapper:v0.5.0
-    commands:
-      - docker build --target test-mods -t k3s:mod -f Dockerfile.test .
-      - docker run -i k3s:mod
-
-    volumes:
-      - name: docker
-        path: /var/run/docker.sock
-volumes:
+- name: skipfiles
+  image: plugins/git
+  commands:
+  - export NAME=$(test $DRONE_BUILD_EVENT = pull_request && echo remotes/origin/${DRONE_COMMIT_BRANCH:-master} || echo ${DRONE_COMMIT_SHA}~)
+  - export DIFF=$(
+      export GREPS=$(for line in $(cat .droneignore); do echo -n " -e $line";done);
+      git --no-pager diff --name-only $NAME | $(echo "grep -v $GREPS")
+    )
+  - if [ -z "$DIFF" ]; then
+        echo "All files in PR are on ignore list";
+        exit 78;
+    else
+        echo "Some files in PR are not ignored, $DIFF";
+    fi;
+
+- name: validate_go_mods
+  image: rancher/dapper:v0.5.0
+  commands:
+  - docker build --target test-mods -t k3s:mod -f Dockerfile.test .
+  - docker run -i k3s:mod
+
+  volumes:
   - name: docker
-    host:
-      path: /var/run/docker.sock
+    path: /var/run/docker.sock
+
+volumes:
+- name: docker
+  host:
+    path: /var/run/docker.sock
```

Comment:
the diff is unclear here, but if you look at the file I just removed some spacing here to conform to the way the rest of the yaml is written (lists have same spacing as parent)